UpsertedProfileOrchestrator/__tests__/handler.test.ts

```diff
@@ -251,4 +254,854 @@ describe("UpsertedProfileOrchestrator", () => {
       }
     );
   });
+
+  it("should call UpdateSubscriptionFeedActivity when profile is created", async () => {
+    const expectedQueueName = "queue_name" as NonEmptyString;
+    const upsertedProfileOrchestratorInput = UpsertedProfileOrchestratorInput.decode(
+      {
+        newProfile: {
+          ...aRetrievedProfile,
+          email: aEmailChanged, // Email changed to start the EmailValidationProcessOrchestrator
+          isInboxEnabled: true // Enable inbox to start the SendWelcomeMessagesActivity
+        },
+        oldProfile: undefined,
+        updatedAt: new Date()
+      }
+    ).getOrElseL(_ =>
+      fail(
+        `Cannot decode UpsertedProfileOrchestratorInput: ${readableReport(_)}`
+      )
+    );
+
+    const emailValidationProcessOrchestratorResult = EmailValidationProcessOrchestratorResult.decode(
+      {
+        kind: "SUCCESS"
+      }
+    ).getOrElseL(_ =>
+      fail(
+        `Cannot decode EmailValidationProcessOrchestratorResult: ${readableReport(
+          _
+        )}`
+      )
+    );
+
+    const sendWelcomeMessagesActivityResult = "SUCCESS";
+
+    const contextMockWithDf = {
+      ...contextMock,
+      df: {
+        Task: {
+          all: (tasks: readonly Task[]) => tasks
+        },
+        callActivityWithRetry: jest
+          .fn()
+          .mockReturnValueOnce(sendWelcomeMessagesActivityResult),
+        callSubOrchestratorWithRetry: jest.fn(
+          () => emailValidationProcessOrchestratorResult
+        ),
+        getInput: jest.fn(() => upsertedProfileOrchestratorInput)
+      }
+    };
+
+    const orchestratorHandler = getUpsertedProfileOrchestratorHandler({
+      notifyOn: fromArray([expectedQueueName]).getOrElseL(undefined),
+      sendCashbackMessage: true
+    })(contextMockWithDf as any);
+
+    const result = orchestratorHandler.next();
```

Comment:
For not calling `next` at each step you can either use `consumeGenerator` as visible here: https://github.com/pagopa/io-functions-pushnotifications/blob/a1506a9330b464b5989e8c119150bd401bca4d8c/utils/durable/utils.ts#L8

```suggestion
    const result = consumeGenerator(orchestratorHandler);
    expect(res).toBe(true);

    expect(contextMockWithDf.df.callActivityWithRetry).toBeCalledWith(
      "UpdateSubscriptionsFeedActivity",
      someRetryOptions,
      {
        fiscalCode: aFiscalCode,
        operation: "SUBSCRIBED",
        subscriptionKind: "PROFILE",
        updatedAt: upsertedProfileOrchestratorInput.updatedAt.getTime(),
        version: upsertedProfileOrchestratorInput.newProfile.version
      }
    );
```